### DISCUSSION TITLE
does ver.3.31 support MacOS Sonoma now?

### DISCUSSION THREAD

**Discussion by kingrhy** [osy86/HaC-Mini#1095] [Releases]
question just as title.  does current version can support Sonoma now?

**Comment by osy** [comment]
There is currently no plans to support macOS Sonoma. As stated last year, this project has retired. If I have some free time, I may work on it or if someone else sends a PR, I can review it.

**Comment by SashonIT** [comment]
I will work on it from mid October on to get Sonoma running on the NUC 8I7HVK.   I will send you then the EFI.  Maybe you can implement this in a USB installer.  

What is with @goodbest  .... He also mentioned couple Months ago, to work on runnin Sonoma on this Maschine.  Is there any Progress? 

I was reading a lot in the German Hackintosh Forum about the Sonoma install Problems.

So i will give it a try in October 2023. 

Stay tuned.

**Reply by kingrhy** [reply]
good news, thank you .

**Comment by ivan-ef** [comment]
> Having issues when trying in AMD Graphics mode. Or atleast that's what I'm assuming is the issue.
> 
> Installs, gets to the booting animation with Apple logo and progress bar but then goes black afterwards and doesn't do anything. Even in safe-mode booting (by holding shift in the picker)

You can try to add  "-v" at boot args. It will help to debug the issues.

**Comment by ivan-ef** [comment]
> I will work on it from mid October on to get Sonoma running on the NUC 8I7HVK. I will send you then the EFI. Maybe you can implement this in a USB installer.
> 
> What is with @goodbest .... He also mentioned couple Months ago, to work on runnin Sonoma on this Maschine. Is there any Progress?
> 
> I was reading a lot in the German Hackintosh Forum about the Sonoma install Problems.
> 
> So i will give it a try in October 2023.
> 
> Stay tuned.

Good to know. It would also be great if you will fork this repo to make your own Sonoma support project.

**Comment by Tom-wgmns** [comment]
Created an Github account to follow this thread :) Looking forward if someone can make it work with Sonoma

**Comment by SashonIT** [comment]
Hey Guys ----> Watch this please.

https://github.com/osy/HaC-Mini/discussions/1091#discussioncomment-7782314

**Comment by immie-jay** [comment]
Could you please send me your EFIImraan Jhetam On 6 Dec 2023, at 23:56, SashonIT ***@***.***> wrote:﻿
Hey Guys ----> Watch this please.
#1091 (reply in thread)

**Reply by SashonIT** [reply]
[Uploading SONOMA 14.1.2 FINAL EFI Kopie.zip…]()
@immie-jay

**Reply by immie-jay** [reply]
link not working

**Reply by SashonIT** [reply]
https://github.com/SashonIT/NUC8I7HVK-SONOMA-14.1-EFI

There you go.

Please remind. you already need to do some steps by your own.

Dont expect, that this is fully working right OOB!

ASk Questions if you Stuck somewhere.

**Reply by sliming** [reply]
LAN2 doesn‘t work

**Reply by SashonIT** [reply]
#############I can confirm this.

###################I dont know why.  It was Running in Ventura until 14.1 or 14.1.2. Since 14.2.1 its not running anymore.

#############################But System Recognizes the first Ethernet. But Internet is not passing through.

!!!! I found the Solution!!!!

Copy appleGB.kext into your KExt Folder of your System EFI. Then the second En0 will work. Which is the First one by the way.

Now both of them are running again.

HEre you can find the https://github.com/osy/HaC-Mini/files/9950812/AppleIGB.kext.zip

Also READ POST 947 of this Repository.
https://github.com/osy/HaC-Mini/issues/947

**Comment by YossiCli** [comment]
Does Anyone try 14.3 with this EFI?

**Reply by SashonIT** [reply]
Yes, Me! Its running on my MAschine Fluent. Without any Problems.

Its also my EFI. So it runs perfect.

Everything is working except of the SD Card Reader.

**Reply by S1999U2** [reply]
So any update on "best practices" for how to update to Sonoma? 😊

**Reply by SashonIT** [reply]
You have to do it Manually.

Create a Sonoma Install Stick manually.

And Copy my Efi afterwards into it.  Then Install.

LAst stable Install is : Sonoma 14.3.1
[efi_Sonoma_14.3.1 .zip](https://github.com/osy/HaC-Mini/files/14422220/efi_Sonoma_14.3.1.zip)


I will upload my LAtest EFI.

But for first INstall you need to Disable the 3 Skywalker KExt.

I have 3 Config.plist in this EFI. 1. Update Config.plist. --> Here are the Skywalker Kexts deactivated
                                                     2. Normal Config.plist                                                     
                                                     3. Config.plist which is currntly used.   Can be on of them above.
Please Note:
Do not Update to 14.4 which will come Soon.   Many Many Problems occurs during the BEta Phase.

I also Deleted my Serial Number.

You have to create one or use your old on.  Keep that in mind.

Also if you have installed a lower Version of Sonoma.  For Updating from the System Preferences you need to Revert the OCLP Patcher first and have to take out all these Skywalker KExt. Then Update. After Update ---> Apply the OCLP PAtch Again and put the 3 Skywalker Kext in again.

**Comment by 1git** [comment]
100% agree **Do not Update to 14.4 which will come Soon. Many Many Problems occurs during the BETA Phase.**
I upgraded to 14.4 on the original EFI/patch from goodbest, and it stops booting... we need time to investigate... Good luck!

**Reply by SashonIT** [reply]
Read this here first. Use deepl Software to translate into your Language. https://www.hackintosh-forum.de/forum/thread/58656-anleitung-f%C3%BCr-broadcom-wlan-patch-unter-macos-sonoma/?postID=788294#post788294

You need this here ….——-> With the corresponding IOSkywalkFamily.kext (vers. 1.1.0) exchange and the new OCLP 1.4.1 root patch, everything worked fine after the update to MacOS 14.4. WLAN is still/again there and works as it should. Thanks to the forum.

Also the OCLP is now updated to Version 1.4.1.

Read this here. https://github.com/dortania/OpenCore-Legacy-Patcher/releases


@1git Please do us a favour and report also your success and how you did it, as only reporting Problems. We also want to read the good and healthy News here.


Also i dont do the 14.4 Update now.   I do it somehow later.  My System is now running Stable and i need it everyday, until i can replace it with a Original M3 MAx System.

**Reply by SashonIT** [reply]
A USer from the Hackintosh forum has just done the update. However, he still had to swap the IOSkywalkFamily-kext separately and then run the root patch of OCLP 1.4.1 again.

**Reply by 1git** [reply]
That did the trick!

I had upgraded so was in an interesting state... 
I edited config.plist to exclude the Skywalker kext...
Booted in about 5 minutes... very slow be patient....

After it booted into 14.4...  (no wifi)
I downloaded and updated Skywalker kext (v1.1.0 see link above)
reboot
booted in 30 seconds...
reapply OCLP patch...
reboot
Booted in 30 seconds 
Wifi is working!

System boots super fast; I'll keep testing and see if I hit bugs...

Thanks for the point in the right direction!

**Reply by SashonIT** [reply]
You are Welcom and Coolio @1git. ….. Congrats …. Thanks for your report.

I faced also long Boottimes in the past……. from 14.1 to 14.2 but not 14.2 to 14.2.1. then again from 14.2.1 to 14.3, but not from 14.3 to 14.3.1

I am glad that you did it. Also glad about to know that 14.4 is running on this little NUC.

Nice to hear, that 14.4 is booting super Fast. 14.2.1 was also super Fast. Since 14.3 and 14.3.1 MY System felt a little slower than ususal.

MAybe i will update the Skywalker Kext to version  1.1.0 before i update.

I will report then if my boottimes are different to yours.

**Comment by SashonIT** [comment]
O.C.L.P Version 1.4.2 came out Yesterday. Just for information!

I Updated now.

35 Minutes in Total.

Here is my 14.4 Sonoma EFI with the updated IOSkywalker.Kext Version 1.1

As always----> I Deleted my Serial Number in Platform Info. You need to put in yours or Generate a new one.

[SONOMA14.4EFI.zip](https://github.com/osy/HaC-Mini/files/14577816/SONOMA14.4EFI.zip)


And yes.... System Response is now snappier then before.

Great

**Reply by dengshenkk** [reply]
Can this EFI be used on macos14.2.1?

**Reply by Deftaudio** [reply]
With this build would you please clarify the status for support of USB BT dongles? I'm using CSR4.0 and see it recognized in the device list, however, it seems it's not initiated properly. 
I understand that your solution is to use Broadcom board with an adapter, but I hope I can run itlwm stable for wifi and just a cheap dongle for BT. Please, suggest.

**Reply by Deftaudio** [reply]
Sorry for checking again, anyone can suggest on USB BT solution?

**Reply by Medizinmann-MD** [reply]
I can report using a CSR 4.0 Dongle from Sonnettech with this EFI and it works.

**Reply by Deftaudio** [reply]
Thanks, I got it running with the original BT/Wifi module.

**Reply by Medizinmann-MD** [reply]
I can report that the Update to 14.5 worked fine and everything seems to work as expected.
BT and WiFi works (using BCM94360NG).

**Comment by Medizinmann-MD** [comment]
> Thanks, I got it running with the original BT/Wifi module.

Great. What Did you add to the EFI?

BTW: What doesn't work for me with this EFI is the BT from my  BCM94360NG - WiFI works - maybe I need to add AirportBrcmFixup - didn't try this so far - BT dongle works for now.

**Comment by Deftaudio** [comment]
I used both airportitlwm and Intel bluetooth from here: https://openintelwireless.github.io/ 
itlwm has an alpha build with Sonoma support. BT needed 4 kexts in total: https://openintelwireless.github.io/IntelBluetoothFirmware/FAQ.html#intelbluetoothfirmware
I can share my package if needed.

**Reply by Medizinmann-MD** [reply]
That would be nice.

Maybe I try my luck with the Intel WiFi/BT as my current solution works but is a bit unstable from time to time.
And I tried fix BT for the my  BCM94360NG without luck - have to stay on the CSR Dongle for now.

**Reply by Deftaudio** [reply]
Here you go: 
https://www.dropbox.com/s/v0ltzmv8b8rp1d8/NUC8_workingconfig.zip?dl=0

**Reply by Medizinmann-MD** [reply]
Thank you very much!

And guess what I saw, when I opened my NUC to change the WiFi-Card…
Well one Antenna – presumable the one for BT (as WiFi worked all the time) - wasn’t affixed quite right anymore (maybe because I was carrying the NUC around a lot lately) – after fixing that BT worked.
Lesson learned (again) – if everything else seems working but doesn’t – it might be also hardware related…sigh…

But nether the less I will try the Intel-WiFi especially as I don't need any continuity features and the WiFi 6 is faster.
